### PR TITLE
Add components for Article

### PR DESCRIPTION
**Pull request by rafayet-monon**
This PR partially does some tickets component part-
- #9 
- #10 
- #11 
- #13 

## What happened

Add components in articles

## Insight

Added components for
- Article list
- Header
- Button
- posting details

## Proof Of Work

Test cases should pass

### CODE REVIEW

File: app/components/base/button/component.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        secondary: SECONDARY_CLASSES,
+        success: SUCCESS_CLASSES,
+        danger: DANGER_CLASSES,
+        link: LINK_CLASSES
+      }.freeze
+
+      def initialize(type: :primary, link: '#', **options)
+        @options = options
+
+        @options[:tag] ||= :a
```
Comment (malparty) [inline app/components/base/button/component.rb:27]: I was wondering a few days ago how to have a clean solution that accepts both `<button>` and `<a>` tags :) 

That way is nice! 
One question though: If the tag is 'button', should we remove the default value of the `link` param?

**Review by malparty — APPROVED** [review]
One minor comment 👆, overall LGTM :) 
Thanks a lot for that one, it gives a much better structure to components~ I'll start from this branch for login Font-End ;-)

**Comment by malparty** [comment]
@rafayet-monon please pull the last commit on your local branch (d1c587d), I've added minor changes so that Unit Tests passes ;-)

File: app/components/base/button/component.rb (hunk 1 above)
Comment (rafayet-monon) [reply app/components/base/button/component.rb:27]: That makes sense. Will do that,

**Review by junan — APPROVED** [review]

**Pull request merged**